engine/renderer/private/renderer.cpp

```diff
@@ -49,12 +52,13 @@ Renderer::Renderer(ApplicationModule& application, const std::shared_ptr<Graphic
     InitializeBloomTargets();
     LoadEnvironmentMap();
 
-    _modelLoader = std::make_unique<ModelLoader>(_context, _ecs);
+    _modelLoader = std::make_unique<ModelLoader>();
 
     _batchBuffer = std::make_shared<BatchBuffer>(_context, 256 * 1024 * 1024, 256 * 1024 * 1024);
+    context->Resources()->MeshResourceManager().SetBatchBuffer(_batchBuffer);
```

Comment:
Hmm, I would think it makes more sense to pass the batch buffer in the create call to a mesh